### PR TITLE
Adding trigger for nightly builds

### PR DESCRIPTION
**Pull request by bikramnehra**
Test pipeline: https://concourse.suse.dev/teams/main/pipelines/bisingh-cap-pre-release

### CODE REVIEW

File: cap-ci/pipeline.yaml.tmpl [hunk 1]
```diff
@@ -68,6 +68,15 @@ resources:
     regexp: {{ $config.s3.regexp }}
     region_name: {{ $config.s3.region }}
 
+{{- if $config.schedule }}
+- name: schedule-interval
+  type: time
+  source:
+    start: {{ $config.schedule.start }}
```
Comment (prabalsharma) [inline cap-ci/pipeline.yaml.tmpl:75]: why not just use 

``` 
 type: time
  source:
    interval: 24h
``` 
?

File: cap-ci/jobs/deploy-kubecf.tmpl [hunk 2]
```diff
@@ -8,13 +8,7 @@
     {{- end }}
   {{- if index .jobs "deploy-stratos" }}
   - get: helm-chart.stratos-chart
-    {{- if not (index .jobs "deploy-k8s") }}
-    trigger: true
```
Comment (prabalsharma) [inline cap-ci/jobs/deploy-kubecf.tmpl]: we still need these triggers for CI using pet clusters

File: cap-ci/jobs/deploy-k8s.tmpl [hunk 3]
```diff
@@ -2,13 +2,17 @@
...
+  - get: schedule-interval
+    trigger: true
+  {{ end }}
   - get: s3.kubecf-bundle
+  {{- if not .schedule }}
     trigger: true
+  {{ end }}
   {{- if index .jobs "deploy-stratos" }}
   - get: helm-chart.stratos-chart
-    trigger: true
```
Comment (prabalsharma) [inline cap-ci/jobs/deploy-k8s.tmpl:9]: we still need these triggers for CI using pet clusters.

File: cap-ci/jobs/deploy-k8s.tmpl [hunk 4]
```diff
@@ -2,13 +2,17 @@
...
   - get: helm-chart.stratos-metrics-chart
-    trigger: true
```
Comment (prabalsharma) [inline cap-ci/jobs/deploy-k8s.tmpl]: we still need these triggers for CI using pet clusters

**Review by prabalsharma — CHANGES_REQUESTED** [review]

File: cap-ci/pipeline.yaml.tmpl (hunk 1 above)
Comment (prabalsharma) [reply cap-ci/pipeline.yaml.tmpl:75]: may we because we want it to be on specific time.  In that case should the interval for check be for 1 hour?

Comment (bikramnehra) [reply cap-ci/pipeline.yaml.tmpl:75]: thanks for pointing I missed the `interval` part. Rest is to ensure we trigger during night time hence `nightly`
https://github.com/concourse/time-resource#trigger-on-an-interval-within-time-range

File: cap-ci/jobs/deploy-kubecf.tmpl (hunk 2 above)
Comment (bikramnehra) [reply cap-ci/jobs/deploy-kubecf.tmpl]: For that, I have kept the trigger on `s3.kubecf-bundle`. I don't think we ever need pipelines to automatically trigger on new stratos charts/releases.

File: cap-ci/jobs/deploy-k8s.tmpl (hunk 3 above)
Comment (bikramnehra) [reply cap-ci/jobs/deploy-k8s.tmpl:9]: same as above.

File: cap-ci/jobs/deploy-k8s.tmpl (hunk 4 above)
Comment (bikramnehra) [reply cap-ci/jobs/deploy-k8s.tmpl]: same as above.

File: cap-ci/cap-pre-release.yaml [hunk 5]
```diff
@@ -49,3 +49,8 @@ s3:
   bucket: kubecf
   region: us-west-2
   regexp: kubecf-bundle-v(.*).tgz
+schedule:
```
Comment (HartS) [inline cap-ci/cap-pre-release.yaml:53]: We should have a boolean flag to control whether scheduling is on/off. `enabled: true`

Also, a comment here like ("if scheduling is enabled, nightlies will be run and triggering will be disabled on ...")

**Review by HartS — CHANGES_REQUESTED** [review]

File: cap-ci/jobs/deploy-kubecf.tmpl (hunk 2 above)
Comment (prabalsharma) [reply cap-ci/jobs/deploy-kubecf.tmpl]: oh we do. cap-release-ci is to test every piece of CAP. Hence, it should get triggered for every new build of its bits. kubecf is not the olny thing this CI is suppose to test. The trigger was added in deliberately.

Comment (bikramnehra) [reply cap-ci/jobs/deploy-kubecf.tmpl]: I am still not fully convinced but I am reverting changes and we can handle it separately.

File: cap-ci/cap-pre-release.yaml (hunk 5 above)
Comment (bikramnehra) [reply cap-ci/cap-pre-release.yaml:53]: DONE.

**Review by HartS — APPROVED** [review]

File: cap-ci/cap-pre-release.yaml [hunk 6]
```diff
@@ -49,3 +49,10 @@ s3:
   bucket: kubecf
   region: us-west-2
   regexp: kubecf-bundle-v(.*).tgz
+# Scheduling is required for nightly builds, enabling it removes regular trigger on kubecf bundle.
+schedule:
+  enabled: true
+  interval: 24h
```
Comment (prabalsharma) [inline cap-ci/cap-pre-release.yaml]: not required

File: cap-ci/cap-pre-release.yaml [hunk 7]
```diff
@@ -49,3 +49,10 @@ s3:
...
+  start: 12:00 AM
+  stop: 1:00 AM
```
Comment (prabalsharma) [inline cap-ci/cap-pre-release.yaml]: make it 10 mins duration

**Pull request merged**